libkbfs/bare_root_metadata_v3.go

```diff
@@ -1013,21 +1040,71 @@ func (md *BareRootMetadataV3) GetTLFReaderKeyBundleID() TLFReaderKeyBundleID {
 }
 
 // FinalizeRekey implements the MutableBareRootMetadata interface for BareRootMetadataV3.
-func (md *BareRootMetadataV3) FinalizeRekey(config Config, extra ExtraMetadata) error {
+func (md *BareRootMetadataV3) FinalizeRekey(crypto cryptoPure, prevKey, currKey TLFCryptKey,
+	extra ExtraMetadata) error {
 	extraV3, ok := extra.(*ExtraMetadataV3)
 	if !ok {
 		return errors.New("Invalid extra metadata")
 	}
+	if (prevKey != TLFCryptKey{}) {
+		numKeys := int(md.LatestKeyGeneration() - 1)
+		var oldKeys []TLFCryptKey
+		var err error
+		if numKeys > 1 {
```

Comment:
I don't think that's enough -- if `md.LatestKeyGeneration()` is `FirstValidKeyGen`, then `numKeys` will be `0`, which is inconsistent with `prevKey` being non-nil.